app/views/posts/show.html.haml

```diff
@@ -0,0 +1,14 @@
+%p
+  %b Body:
+  = @post.body
+%p
+  %b Tagged under:
+  = @post.tag_list
+    
+
+%b Comments:
+= render partial: 'comments/comment', collection: @post.comments
+
+= link_to 'Leave a Comment', new_post_comment_path(@post)
```

Comment:
Let's say a requirement was added so that comments need to be added directly from the post page, without a page reload or going to a different page to write the comment. How would you approach that? (Again, no need to write the code, just jot down how you'd go about it.)

Reply:
I think you'll find that using `Comment.new` in the form here has problems in that if the comment fails validation for whatever reason, the user's comment will essentially be erased when the form re-renders. 

Otherwise, yup, makes sense!

Reply:
Your right.

Alternatively we can 
```
= render partial: 'comments/form', locals: {post: @post, comment: @comment ||= Comment.new }
```

With this @comment will be a new empty instance when the user first visits the Post #show page, then if there is an error the @comment instance will be the instance from the Comments#create action which holds the form values.

This way value will be show in the form on validation error.